src/sage/schemes/curves/plane_curve_arrangement.py

```diff
@@ -0,0 +1,1246 @@
+        C_affine = affine(affines)
+        proj = not (infinity_divides or infinity_in_C)
+        G = C_affine.fundamental_group(simplified=simplified, vertical=True,
+                                       projective=proj)
+        dic = C_affine.meridians(simplified=simplified, vertical=True)
+        if infinity_in_C:
+            dic1 = dict()
+            for k in range(j):
+                dic1[k] = dic[k]
+            dic1[j] = dic[n - 1]
+            for k in range(j + 1, n):
+                dic1[k] = dic[k - 1]
+        elif infinity_divides:
+            dic1 = {k: dic[k] for k in range(n)}
+            dic1[j] += dic[n]
+        else:
+            dic1 = dic
+        if not simplified:
+            self._fundamental_group = G
+            self._meridians = dic1
+        else:
+            self._fundamental_group_simplified = G
+            self._meridians_simplified = dic1
+        return G
+
+    def meridians(self, simplified=True):
+        r"""
+        Meridians of each irreducible component.
+
+        OUTPUT:
+
+        A dictionary which associates the index of each curve with
+        its meridians, including the line at infinity if it can be
+        computed
+
+        EXAMPLES::
+
+            sage: # needs sirocco
+            sage: H.<x, y, z> = ProjectivePlaneCurveArrangements(QQ)
+            sage: A = H(y^2 + x*z, y + x - z, x)
+            sage: A.fundamental_group().sorted_presentation()
+            Finitely presented group < x0, x1 | x1^-1*x0^-1*x1*x0 >
+            sage: A.meridians()
+            {0: [x1], 1: [x0], 2: [x1^-1*x0^-1*x1^-1]}
+            sage: A = H(y^2 + x*z, z, x)
+            sage: A.fundamental_group()
+            Finitely presented group < x0, x1 | (x1*x0)^2*(x1^-1*x0^-1)^2 >
+            sage: A.meridians()
+            {0: [x0, x1*x0*x1^-1], 1: [x0^-1*x1^-1*x0^-1], 2: [x1]}
+            sage: A = H(y^2 + x*z, z*x, y)
+            sage: A.fundamental_group()
+            Finitely presented group < x0, x1, x2 | x2*x0*x1*x0^-1*x2^-1*x1^-1,
+                                                    x1*(x2*x0)^2*x2^-1*x1^-1*x0^-1*x2^-1*x0^-1 >
+            sage: A.meridians()
+            {0: [x0, x2*x0*x2^-1], 1: [x2, x0^-1*x2^-1*x1^-1*x0^-1], 2: [x1]}
+
+        This function needs
+        :func:`ProjectivePlaneCurveArrangements.fundamental_group`
+        with the same options, where some examples are shown.
+
+        .. WARNING::
+
+            This functionality requires the sirocco package to be installed.
+        """
+        if not simplified:
+            computed = bool(self._meridians)
+        else:
+            computed = bool(self._meridians_simplified)
+        if not computed:
+            _ = self._fundamental_group(simplified=simplified)
+        if not simplified:
+            return self._meridians
+        return self._meridians_simplified
+
+
+class AffinePlaneCurveArrangements(Parent, UniqueRepresentation):
```

Comment:
```suggestion
class AffinePlaneCurveArrangements(UniqueRepresentation, Parent):
```
so the comparison and hashing is guaranteed inherited from `UniqueRepresentation`.